src/Microsoft.VisualStudio.Sdk.TestFramework.Xunit/BrokeredServiceContractTestBase`3.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.IO.Pipelines;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.VisualStudio.Threading;
+using Nerdbank.Streams;
+using Xunit;
+
+namespace Microsoft.VisualStudio.Sdk.TestFramework
+{
+    /// <summary>
+    /// A base class for testing Visual Studio brokered service contracts with client callbacks.
+    /// </summary>
+    /// <typeparam name="TInterface">The service interface.</typeparam>
+    /// <typeparam name="TServiceMock">The class that mocks the service.</typeparam>
+    /// <typeparam name="TClientInterfaceMock">The class that mockes the service's client callback.</typeparam>
+    public abstract class BrokeredServiceContractTestBase<TInterface, TServiceMock, TClientInterfaceMock> : BrokeredServiceContractTestBase<TInterface, TServiceMock>
+        where TInterface : class
+        where TServiceMock : TInterface, IMockServiceWithClientCallback, new()
```

Comment:
👏🏼 I like the additional type constraint you came up with here. That should help people realize how they need to work with this.